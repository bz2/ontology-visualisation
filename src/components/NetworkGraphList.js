import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { Button } from 'primereact/button'
import actions from '../store/actions'
import { SIDEBAR_VIEW_GRAPHS } from '../constants/views'

const NetworkGraphList = ({
  setStoreState,
  removeFromObject,
  currentGraph,
  graphData
}) => {
  const { t } = useTranslation()

  const graphViewsKeys = Object.keys(graphData)

  return (
    <>
      <div className="sidebar-main-title">
        {t(SIDEBAR_VIEW_GRAPHS)}
      </div>
      <div className="network-graph-list">
        {
          graphViewsKeys.map((graphViewsKey) => {
            const { label, noDelete } = graphData[graphViewsKey]

            return (
              <div
                className={`network-graph-list-row ${currentGraph === graphViewsKey ? 'network-graph-list-row-selected' : ''}`}
                key={`selected-node-row-${graphViewsKey}`}
              >
                <div className="network-graph-list-row-delete">
                  {
                    !noDelete && (
                      <Button
                        aria-label={t('removeGraph')}
                        tooltip={`${t('removeGraph')}: ${label}`}
                        onClick={() => {
                          if (currentGraph === graphViewsKey) {
                            setStoreState('currentGraph', 'graph-0')
                          }

                          removeFromObject('graphData', graphViewsKey)
                        }}
                        icon="pi pi-times"
                      />
                    )
                  }
                </div>

                <div className="network-graph-list-row-main">
                  <Button
                    aria-label={t('viewGraph')}
                    tooltip={`${t('viewGraph')}: ${label}`}
                    disabled={currentGraph === graphViewsKey}
                    onClick={() => setStoreState('currentGraph', graphViewsKey)}
                    label={label}
                    icon="pi pi-chevron-right"
                    iconPos="right"
                  />
                </div>
              </div>
            )
          })
        }
      </div>
    </>
  )
}

NetworkGraphList.propTypes = {
  setStoreState: PropTypes.func.isRequired,
  graphData: PropTypes.shape().isRequired,
  removeFromObject: PropTypes.func.isRequired,
  currentGraph: PropTypes.string.isRequired,
}

const mapToProps = ({
  graphData,
  currentGraph
}) => ({
  graphData,
  currentGraph
})

export default connect(
  mapToProps,
  actions
)(NetworkGraphList)
