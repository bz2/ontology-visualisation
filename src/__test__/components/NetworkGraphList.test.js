import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NetworkGraphList from '../../components/NetworkGraphList'

const setup = () => {
  const props = {
    currentGraph: 'graph-0',
    graphData: {
      'graph-0': {
        label: 'Main',
        noDelete: true
      },
      'graph-1': {
        label: 'Search',
      }
    },
    updateStoreValue: jest.fn(),
  }

  const component = shallow(<NetworkGraphList {...props} />)

  return {
    component,
    props
  }
}

describe('NetworkGraphList', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
