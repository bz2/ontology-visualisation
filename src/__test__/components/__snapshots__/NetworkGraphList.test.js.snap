// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkGraphList should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  currentGraph="graph-0"
  graphData={
    Object {
      "graph-0": Object {
        "label": "Main",
        "noDelete": true,
      },
      "graph-1": Object {
        "label": "Search",
      },
    }
  }
  mapToProps={[Function]}
  removeFromObject={[MockFunction]}
  setStoreState={[MockFunction]}
>
  <Component />
</Connect>
`;
